ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -221,167 +210,123 @@ public static Object toFieldType(final Expression expressionValue) {
   */
   // CHECKSTYLE_RULES.OFF: CyclomaticComplexity
   public static SqlCommand transformToSqlCommand(
-      // CHECKSTYLE_RULES.ON: CyclomaticComplexity
-      final String sql, final Map<String, String> variables) {
+          // CHECKSTYLE_RULES.ON: CyclomaticComplexity
+          final String sql, final Map<String, String> variables) {
 
-    // Splits the sql string into tokens(uppercased keyowords, identifiers and strings)
-    final List<String> tokens = Arrays
-        .stream(sql.toUpperCase().split(QUOTED_STRING_OR_WHITESPACE))
-        .filter(s -> !s.isEmpty())
-        .collect(Collectors.toList());
-    switch (getStatementType(tokens)) {
+    validateSupportedStatementType(sql);
+    final String substituted;
+    try {
+      substituted = VariableSubstitutor.substitute(KSQL_PARSER.parse(sql).get(0), variables);
```

Comment:
Since we're substituting variables here, we can stop passing variables to the ksql client calls in `ApplyMigrationCommand`.

Reply:
In [executeCommand](https://github.com/confluentinc/ksql/blob/f2e4991d043f05f152714e9ceb495619b0617917/ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java#L406), add/remove the variable to the the [map](https://github.com/confluentinc/ksql/blob/f2e4991d043f05f152714e9ceb495619b0617917/ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java#L371) of variables instead of calling the Java client. Same thing with [setUpJavaClientVariables](https://github.com/confluentinc/ksql/blob/f2e4991d043f05f152714e9ceb495619b0617917/ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java#L390).